Dockerfile

```diff
@@ -1,11 +1,16 @@
-FROM golang:1.16.0 AS BUILDER
+FROM --platform=$TARGETPLATFORM golang:1.16.0 AS BUILDER
+
+ARG TARGETPLATFORM
+ARG TARGETARCH
+ARG BUILDPLATFORM
+
 WORKDIR /go/src/github.com/jtblin/kube2iam
 ENV ARCH=linux
 ENV CGO_ENABLED=0
 COPY . ./
 RUN make setup && make build
 
-FROM alpine:3.14.3
+FROM --platform=$TARGETPLATFORM alpine:3.14.3
```

Comment:
@rr-nick-tan I think you can remove all your changes in this file. When the image is built by buildx it should automatically pick up the correct variant for the target platform. Maybe you can give it a try?